{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "5667639a-83d5-4716-a9a7-14f681127234",
   "metadata": {
    "tags": []
   },
   "source": [
    "## ProteoQ analysis for Van Tine et al manuscript\n",
    "* search results for this analysis are in the vanTine_001_90_samples.zip file of MassIVE MSV000092188, vanTine_001_90_samples.zip. \n",
    "* Save them locally in the working directory."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "593fbfc7-2a9b-4411-8407-fa6842e45330",
   "metadata": {
    "tags": [],
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Registered S3 method overwritten by 'GGally':\n",
      "  method from   \n",
      "  +.gg   ggplot2\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Welcome to proteoQ.\n",
      "\n",
      "=============================================================================\n",
      "NEW features (v1.7.5.1):\n",
      "[x] Database searches: https://github.com/qzhang503/mzion\n",
      "=============================================================================\n",
      "\n",
      "\n",
      "Loading required package: registry\n",
      "\n",
      "Registered S3 methods overwritten by 'registry':\n",
      "  method               from \n",
      "  print.registry_field proxy\n",
      "  print.registry_entry proxy\n",
      "\n",
      "Loading required package: rngtools\n",
      "\n",
      "Loading required package: cluster\n",
      "\n",
      "NMF - BioConductor layer [OK] | Shared memory capabilities [NO: bigmemory] | Cores 2/2\n",
      "\n",
      "  To enable shared memory capabilities, try: install.extras('\n",
      "NMF\n",
      "')\n",
      "\n",
      "── \u001b[1mAttaching core tidyverse packages\u001b[22m ──────────────────────────── tidyverse 2.0.0 ──\n",
      "\u001b[32m✔\u001b[39m \u001b[34mdplyr    \u001b[39m 1.1.2     \u001b[32m✔\u001b[39m \u001b[34mreadr    \u001b[39m 2.1.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mforcats  \u001b[39m 1.0.0     \u001b[32m✔\u001b[39m \u001b[34mstringr  \u001b[39m 1.5.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mlubridate\u001b[39m 1.9.2     \u001b[32m✔\u001b[39m \u001b[34mtibble   \u001b[39m 3.2.1\n",
      "\u001b[32m✔\u001b[39m \u001b[34mpurrr    \u001b[39m 1.0.1     \u001b[32m✔\u001b[39m \u001b[34mtidyr    \u001b[39m 1.3.0\n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mcombine()\u001b[39m    masks \u001b[34mBiobase\u001b[39m::combine(), \u001b[34mBiocGenerics\u001b[39m::combine()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m     masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m        masks \u001b[34mstats\u001b[39m::lag()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mggplot2\u001b[39m::\u001b[32mPosition()\u001b[39m masks \u001b[34mBiocGenerics\u001b[39m::Position(), \u001b[34mbase\u001b[39m::Position()\n",
      "\u001b[36mℹ\u001b[39m Use the conflicted package (\u001b[3m\u001b[34m<http://conflicted.r-lib.org/>\u001b[39m\u001b[23m) to force all conflicts to become errors\n",
      "\n",
      "Attaching package: ‘MASS’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:dplyr’:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘mice’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:BiocGenerics’:\n",
      "\n",
      "    cbind, rbind\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    filter\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    cbind, rbind\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(proteoQDA) #devtools::install_github(\"qzhang503/proteoQDA\")\n",
    "library(proteoQ) #devtools::install_github(\"qzhang503/proteoQ\")\n",
    "library(NMF) #install.packages('NMF')\n",
    "library(downloader) #install.packages('downloader')\n",
    "library(RColorBrewer) #install.packages('RColorBrewer')\n",
    "library(pheatmap) #install.packages('pheatmap')\n",
    "library(ggplot2) #install.packages('ggplot2')\n",
    "library(tidyverse) #install.packages('tidyverse')\n",
    "library(ggthemes) #install.packages('ggthemes')\n",
    "library(MASS) #install.packages('MASS')\n",
    "library(ggrepel) #install.packages('ggrepel')\n",
    "library(mice)  #install.packages('mice')\n",
    "`%notin%` <- Negate(`%in%`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0bd9702e-ba1a-4864-b7ba-0a7f4cefbd1c",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "R version 4.2.1 (2022-06-23)\n",
       "Platform: x86_64-apple-darwin17.0 (64-bit)\n",
       "Running under: macOS Big Sur ... 10.16\n",
       "\n",
       "Matrix products: default\n",
       "BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib\n",
       "LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib\n",
       "\n",
       "locale:\n",
       "[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8\n",
       "\n",
       "attached base packages:\n",
       "[1] stats     graphics  grDevices utils     datasets  methods   base     \n",
       "\n",
       "other attached packages:\n",
       " [1] mice_3.15.0         ggrepel_0.9.3       MASS_7.3-58.3      \n",
       " [4] ggthemes_4.2.4      lubridate_1.9.2     forcats_1.0.0      \n",
       " [7] stringr_1.5.0       dplyr_1.1.2         purrr_1.0.1        \n",
       "[10] readr_2.1.4         tidyr_1.3.0         tibble_3.2.1       \n",
       "[13] tidyverse_2.0.0     ggplot2_3.4.2       pheatmap_1.0.12    \n",
       "[16] RColorBrewer_1.1-3  downloader_0.4      NMF_0.26           \n",
       "[19] Biobase_2.58.0      BiocGenerics_0.44.0 cluster_2.1.4      \n",
       "[22] rngtools_1.5.2      registry_0.5-1      proteoQ_1.7.5.1    \n",
       "[25] proteoQDA_0.0.2.0  \n",
       "\n",
       "loaded via a namespace (and not attached):\n",
       " [1] colorspace_2.1-0       class_7.3-21           IRdisplay_1.1         \n",
       " [4] XVector_0.38.0         base64enc_0.1-3        fs_1.6.1              \n",
       " [7] proxy_0.4-27           bit64_4.0.5            AnnotationDbi_1.60.2  \n",
       "[10] fansi_1.0.4            codetools_0.2-19       splines_4.2.1         \n",
       "[13] doParallel_1.0.17      cachem_1.0.7           IRkernel_1.3.2        \n",
       "[16] jsonlite_1.8.4         broom_1.0.4            gridBase_0.4-7        \n",
       "[19] kernlab_0.9-32         png_0.1-8              BiocManager_1.30.20   \n",
       "[22] compiler_4.2.1         httr_1.4.5             backports_1.4.1       \n",
       "[25] Matrix_1.5-4           fastmap_1.1.1          lazyeval_0.2.2        \n",
       "[28] limma_3.54.2           cli_3.6.1              org.Rn.eg.db_3.16.0   \n",
       "[31] org.Mm.eg.db_3.16.0    htmltools_0.5.5        tools_4.2.1           \n",
       "[34] gtable_0.3.3           glue_1.6.2             GenomeInfoDbData_1.2.9\n",
       "[37] reshape2_1.4.4         fastmatch_1.1-3        Rcpp_1.0.10           \n",
       "[40] cellranger_1.1.0       vctrs_0.6.2            Biostrings_2.66.0     \n",
       "[43] nlme_3.1-162           iterators_1.0.14       openxlsx_4.2.5.2      \n",
       "[46] timechange_0.2.0       lifecycle_1.0.3        org.Hs.eg.db_3.16.0   \n",
       "[49] zlibbioc_1.44.0        scales_1.2.1           hms_1.1.3             \n",
       "[52] parallel_4.2.1         memoise_2.0.1          segmented_1.6-3       \n",
       "[55] reshape_0.8.9          stringi_1.7.12         RSQLite_2.3.1         \n",
       "[58] S4Vectors_0.36.2       foreach_1.5.2          e1071_1.7-13          \n",
       "[61] zip_2.2.2              repr_1.1.6             GenomeInfoDb_1.34.9   \n",
       "[64] rlang_1.1.0            pkgconfig_2.0.3        bitops_1.0-7          \n",
       "[67] evaluate_0.20          lattice_0.20-45        htmlwidgets_1.6.2     \n",
       "[70] bit_4.0.5              tidyselect_1.2.0       GGally_2.1.2          \n",
       "[73] plyr_1.8.8             magrittr_2.0.3         R6_2.5.1              \n",
       "[76] IRanges_2.32.0         generics_0.1.3         pbdZMQ_0.3-9          \n",
       "[79] DBI_1.1.3              pillar_1.9.0           withr_2.5.0           \n",
       "[82] mixtools_2.0.0         survival_3.5-5         KEGGREST_1.38.0       \n",
       "[85] RCurl_1.98-1.12        crayon_1.5.2           uuid_1.1-0            \n",
       "[88] utf8_1.2.3             plotly_4.10.1          tzdb_0.3.0            \n",
       "[91] grid_4.2.1             readxl_1.4.2           data.table_1.14.8     \n",
       "[94] blob_1.2.4             digest_0.6.31          stats4_4.2.1          \n",
       "[97] munsell_0.5.0          viridisLite_0.4.1     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sessionInfo()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ccb87f43-fef4-4d91-a1e3-5703bed56d8c",
   "metadata": {
    "tags": [],
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "'/Users/jasonheld/Manuscripts/2022_Sarcoma-Exosomes/vanTine_001_FinalDataWithControls/MQ/vanTine_001_90_samples'"
      ],
      "text/latex": [
       "'/Users/jasonheld/Manuscripts/2022\\_Sarcoma-Exosomes/vanTine\\_001\\_FinalDataWithControls/MQ/vanTine\\_001\\_90\\_samples'"
      ],
      "text/markdown": [
       "'/Users/jasonheld/Manuscripts/2022_Sarcoma-Exosomes/vanTine_001_FinalDataWithControls/MQ/vanTine_001_90_samples'"
      ],
      "text/plain": [
       "[1] \"/Users/jasonheld/Manuscripts/2022_Sarcoma-Exosomes/vanTine_001_FinalDataWithControls/MQ/vanTine_001_90_samples\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "'/Library/Frameworks/R.framework/Versions/4.2/Resources/library'"
      ],
      "text/latex": [
       "'/Library/Frameworks/R.framework/Versions/4.2/Resources/library'"
      ],
      "text/markdown": [
       "'/Library/Frameworks/R.framework/Versions/4.2/Resources/library'"
      ],
      "text/plain": [
       "[1] \"/Library/Frameworks/R.framework/Versions/4.2/Resources/library\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "setwd(\"/Users/jasonheld/Manuscripts/2022_Sarcoma-Exosomes/vanTine_001_FinalDataWithControls/MQ/vanTine_001_90_samples\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "effe4ec5-8d82-410a-8de7-36b99e062888",
   "metadata": {
    "tags": [],
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "# serum genes to filter\n",
    "\n",
    "serumGenes <- c(\n",
    "  \"SERPINA1\",\n",
    "  \"HP\",\n",
    "  \"IGHA1\",\n",
    "  \"IGHG1\",\n",
    "  \"FGA\",\n",
    "  \"ALB\",\n",
    "  \"TF\",\n",
    "  \"ORM1\",\n",
    "  \"A2M\",\n",
    "  \"APOA1\",\n",
    "  \"APOA2\",\n",
    "  \"C3\",\n",
    "  \"IGHM\",\n",
    "  \"TTR\",\n",
    "  \"HBD\"\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b184bb84-454d-4781-97d3-906e9975ec81",
   "metadata": {
    "tags": [],
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\u001b[1mRows: \u001b[22m\u001b[34m30298\u001b[39m \u001b[1mColumns: \u001b[22m\u001b[34m418\u001b[39m\n",
      "\u001b[36m──\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m────────────────────────────────────────────────────────────\u001b[39m\n",
      "\u001b[1mDelimiter:\u001b[22m \"\\t\"\n",
      "\u001b[31mchr\u001b[39m (114): Sequence, N-term cleavage window, C-term cleavage window, Amino a...\n",
      "\u001b[32mdbl\u001b[39m (303): A Count, R Count, N Count, D Count, C Count, Q Count, E Count, G ...\n",
      "\u001b[33mlgl\u001b[39m   (1): Taxonomy IDs\n",
      "\n",
      "\u001b[36mℹ\u001b[39m Use `spec()` to retrieve the full column specification for this data.\n",
      "\u001b[36mℹ\u001b[39m Specify the column types or set `show_col_types = FALSE` to quiet this message.\n"
     ]
    }
   ],
   "source": [
    "# load_expts\n",
    "load_expts(\n",
    "    \"/Users/jasonheld/Manuscripts/2022_Sarcoma-Exosomes/vanTine_001_FinalDataWithControls/MQ/vanTine_001_90_samples\"\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "be5fca73-13d0-4ed7-9436-9f14bd1ba055",
   "metadata": {
    "tags": [],
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Primary column keys in \"msms[...].txt\" for \"filter_\" varargs.\n",
      "\n",
      "Warning message:\n",
      "“\n",
      "================================================================\n",
      "Column keys in \"msms[...].txt\" may be changed to a \"Title Case\":\n",
      "( ) \"Protein Names\" <-> \"Protein Names\"... \n",
      "(x) \"Protein names\" -> \"Protein Names\"... \n",
      "(Lookups in \"~/PSM/cache/MQ_colkey_lookup.csv\".)\n",
      "================================================================\n",
      "”\n",
      "At acc_type = \"uniprot_acc\":\n",
      " the number of entries in FASTA: 20375\n",
      " the number of entries matched: 3534\n",
      " percent annotated: 0.173\n",
      "\n",
      "\n",
      "Parsing shared proteins...\n",
      "\n",
      "\n",
      "At acc_type = \"other_acc\":\n",
      " the number of entries in FASTA: 20375\n",
      " the number of entries matched: 0\n",
      " percent annotated: 0.000\n",
      "\n",
      "At acc_type = \"uniprot_acc\":\n",
      " the number of entries in FASTA: 20375\n",
      " the number of entries matched: 3948\n",
      " percent annotated: 0.194\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n",
      "No intensity columns available.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "normPSM(\n",
    "  #col_select = 'LMS', # this selection is not useful in the end. Don't be tempted\n",
    "  group_psm_by = pep_seq, \n",
    "  group_pep_by = gene, \n",
    "  fasta = c(\"/Users/jasonheld/Manuscripts/2022_Sarcoma-Exosomes/vanTine_001_FinalDataWithControls/uniprot_hs_2020_05.fasta\"), \n",
    "  rm_craps = TRUE,\n",
    "  rm_krts = FALSE,\n",
    "  rm_outliers = FALSE, \n",
    "  annot_kinases = TRUE, \n",
    "  plot_rptr_int = TRUE, \n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fbf83bbc-096f-4da9-98ba-07e9da1181b3",
   "metadata": {
    "tags": [],
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Primary column keys in \"PSM/TMTset1_LCMSinj1_PSM_N.txt\" etc. for \"filter_\" varargs.\n",
      "\n"
     ]
    },
    {
     "ename": "ERROR",
     "evalue": "Error in checkForRemoteErrors(val): 89 nodes produced errors; first error: No samples at TMT_Set 2 and LCMS_Injection 2.\n",
     "output_type": "error",
     "traceback": [
      "Error in checkForRemoteErrors(val): 89 nodes produced errors; first error: No samples at TMT_Set 2 and LCMS_Injection 2.\nTraceback:\n",
      "1. PSM2Pep()",
      "2. suppressWarnings(silent_out <- parallel::clusterApply(cl, filelist, \n .     psm_to_pep, dat_dir = dat_dir, label_scheme_full = label_scheme_full, \n .     group_psm_by = group_psm_by, group_pep_by = group_pep_by, \n .     method_psm_pep = method_psm_pep, lfq_ret_tol = lfq_ret_tol, \n .     rm_allna = rm_allna, type_sd = type_sd, ...))",
      "3. withCallingHandlers(expr, warning = function(w) if (inherits(w, \n .     classes)) tryInvokeRestart(\"muffleWarning\"))",
      "4. parallel::clusterApply(cl, filelist, psm_to_pep, dat_dir = dat_dir, \n .     label_scheme_full = label_scheme_full, group_psm_by = group_psm_by, \n .     group_pep_by = group_pep_by, method_psm_pep = method_psm_pep, \n .     lfq_ret_tol = lfq_ret_tol, rm_allna = rm_allna, type_sd = type_sd, \n .     ...)",
      "5. staticClusterApply(cl, fun, length(x), argfun)",
      "6. checkForRemoteErrors(val)",
      "7. stop(count, \" nodes produced errors; first error: \", firstmsg, \n .     domain = NA)"
     ]
    }
   ],
   "source": [
    "# PSMs to peptides\n",
    "#?PSM2Pep # help\n",
    "PSM2Pep()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b7f9c83c-490e-4142-8662-d9dc19726d05",
   "metadata": {
    "tags": [],
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "# peptide data merging\n",
    "mergePep(\n",
    "  # omit_single_lfq = FALSE, \n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3e0a15e1-3c28-4512-914a-8c7b85388a1e",
   "metadata": {
    "tags": [],
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "# peptide data alignment\n",
    "\n",
    "standPep(\n",
    "  method_align = MGKernel, #MGKernal, #MC\n",
    "  range_log2r = c(5, 95), \n",
    "  range_int = c(5, 95), \n",
    "  n_comp = 3, # defines the number of Gaussian kernels\n",
    "  seed = 749662, # set a seed for reproducible fittings\n",
    "  maxit = 200, \n",
    "  epsilon = 1e-05, \n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e10f3693-1d03-48cf-872c-a38e9aa5f96d",
   "metadata": {
    "tags": [],
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "# cleanup peptide groups\n",
    "purgePep(\n",
    "    filename = purgePEP.png, # Peptide/log2FC_cv/purged\n",
    "    ymax = 4 # for the plot\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "17c7206e-b8a9-4ba7-a2ab-86df4deffe08",
   "metadata": {
    "tags": [],
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "# peptides to proteins\n",
    "\n",
    "Pep2Prn(\n",
    "    use_unique_pep = TRUE,\n",
    "    filter_topserum = exprs(gene %notin% serumGenes),\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8a674d0c-0daf-4128-9c10-393de6108e2d",
   "metadata": {
    "tags": [],
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "# protein data standardization by human subsets and \n",
    "# with a minimal of three measures (out of six) in intensity\n",
    "standPrn(\n",
    "  method_align = MGKernel, #MGKernel, #MC\n",
    "  range_log2r = c(5, 95), \n",
    "  range_int = c(5, 95), \n",
    "  n_comp = 3, \n",
    "  seed = 749662, \n",
    "  maxit = 200, \n",
    "  epsilon = 1e-05, \n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4b844b5b-326d-4f68-bcdf-d98b4dbff043",
   "metadata": {
    "tags": [],
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "## scale\n",
    "scale_log2r = TRUE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e740f2ba-f7d9-4b37-a4bf-da0dcea60dae",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Primary column keys in `Model/Peptide[_impNA_pVals].txt` for `filter_` varargs.\n",
      "\n",
      "Primary file loaded: /Users/jasonheld/Manuscripts/2022_Sarcoma-Exosomes/vanTine_001_FinalDataWithControls/MQ/vanTine_001_90_samples/Peptide/Model/Peptide_pVals.txt\n",
      "\n",
      "\n",
      "scale_log2r = TRUE\n",
      "\n",
      "impute_na = FALSE\n",
      "\n",
      "complete_cases = FALSE\n",
      "\n",
      "Warning message:\n",
      "“columns with all-NA values for aesthetics excluded: \n",
      "TMT_Channel, Order, Fill, Size, Alpha, Duplicate, Benchmark, Viability, Description.\n",
      "”\n",
      "Warning message:\n",
      "“Columns at a single-factor level for aesthetics excluded: \n",
      "TMT_Set, LCMS_Injection, Reference.\n",
      "”\n",
      "Primary column keys in `Model/Peptide[_impNA_pVals].txt` for `filter_` varargs.\n",
      "\n",
      "Primary file loaded: /Users/jasonheld/Manuscripts/2022_Sarcoma-Exosomes/vanTine_001_FinalDataWithControls/MQ/vanTine_001_90_samples/Peptide/Model/Peptide_pVals.txt\n",
      "\n",
      "\n",
      "scale_log2r = TRUE\n",
      "\n",
      "impute_na = FALSE\n",
      "\n",
      "complete_cases = FALSE\n",
      "\n",
      "Warning message:\n",
      "“columns with all-NA values for aesthetics excluded: \n",
      "TMT_Channel, Order, Fill, Size, Alpha, Duplicate, Benchmark, Viability, Description.\n",
      "”\n",
      "Warning message:\n",
      "“Columns at a single-factor level for aesthetics excluded: \n",
      "TMT_Set, LCMS_Injection, Reference.\n",
      "”\n",
      "\u001b[1m\u001b[22mSaving 7 x 7 in image\n",
      "\u001b[1m\u001b[22mSaving 7 x 7 in image\n",
      "\u001b[1m\u001b[22mSaving 7 x 7 in image\n",
      "Warning message in geom_point(aes(type = \"t\", level = 0.9, colour = Subtype, shape = Timing), :\n",
      "“\u001b[1m\u001b[22mIgnoring unknown aesthetics: \u001b[32mtype\u001b[39m and \u001b[32mlevel\u001b[39m”\n",
      "\u001b[1m\u001b[22mSaving 7 x 7 in image\n"
     ]
    }
   ],
   "source": [
    "# I think the is the final code\n",
    "### MDS\n",
    "# peptide\n",
    "\n",
    "pepMDS(\n",
    "    show_ids = FALSE,\n",
    "    show_ellipses = FALSE,\n",
    "    #col_select=Timing, # doesn't seem to work\n",
    "    filter_ = exprs(count_nna >= 30), # 60 is heatmap # require this many samples to have the protein detected\n",
    "    filter_ = exprs(prot_n_pep >= 2), # 2 is heatmap\n",
    "#    col_select = Subtype, # optional, only works if a subset of rows is annotated\n",
    "    dimension = 3,\n",
    "    col_color = Subtype, # Sensitivity #Cancer, Subtype\n",
    "    color_brewer = Set1,\n",
    "    col_shape = Timing, #Sensitivity, Cancer, Timing\n",
    "#    col_size = Cancer,\n",
    "    method = euclidean, # \"euclidean\", \"maximum\", \"manhattan\", \"canberra\", \"binary\"\n",
    "#    adjEucDist = TRUE, # default is FALSE\n",
    "    width = 7,\n",
    "    height = 7,\n",
    "    filename = Peptide_MDS_Z.pdf # Protein_MDS_Z.png is default\n",
    ")\n",
    "\n",
    "## ggplot 3 dimensions, manually bring together\n",
    "\n",
    "res <- pepMDS(filter_ = exprs(count_nna >= 65))\n",
    "\n",
    "p_peptideMDS <- ggplot(res, aes(x = Coordinate.1, y = Coordinate.2)) +\n",
    "  geom_point(aes(colour = Subtype, shape = Timing), size = 4, stroke = 0.02) + \n",
    "  scale_alpha_manual(values = c(.5, .9)) + \n",
    "  stat_ellipse(type = \"t\", level = 0.90, aes(fill = Cancer), geom = \"polygon\", alpha = .2) + \n",
    "  guides(fill = \"none\") + \n",
    "  labs(title = \"\", x = \"Coordinate 1\", y = \"Coordinate 2\") +\n",
    "  theme_minimal() # theme_minimal(), theme_tufte() +\n",
    "  #xlim(-140,60) + ylim(-40,40) # will need to match based on whatever the best results look like\n",
    "\n",
    "ggsave(file.path(dat_dir, \"Peptide/MDS/peptide_MDS_ggplot_Coord1-2.pdf\"))\n",
    "\n",
    "p_peptideMDS <- ggplot(res, aes(x = Coordinate.1, y = Coordinate.3)) +\n",
    "  geom_point(aes(colour = Subtype, shape = Timing), size = 4, stroke = 0.02) + \n",
    "  scale_alpha_manual(values = c(.5, .9)) + \n",
    "  stat_ellipse(type = \"t\", level = 0.90, aes(fill = Cancer), geom = \"polygon\", alpha = .2) + \n",
    "  guides(fill = \"none\") + \n",
    "  labs(title = \"\", x = \"Coordinate 1\", y = \"Coordinate 3\") +\n",
    "  theme_minimal() # theme_minimal(), theme_tufte() +\n",
    "  #xlim(-140,60) + ylim(-35,35) # will need to match based on whatever the best results look like\n",
    "\n",
    "ggsave(file.path(dat_dir, \"Peptide/MDS/peptide_MDS_ggplot_Coord1-3.pdf\"))\n",
    "\n",
    "p_peptideMDS <- ggplot(res, aes(x = Coordinate.2, y = Coordinate.3)) +\n",
    "  geom_point(aes(type = \"t\", level = 0.90, colour = Subtype, shape = Timing), size = 4, stroke = 0.02) + \n",
    "  scale_alpha_manual(values = c(.5, .9)) + \n",
    "  stat_ellipse(aes(fill = Cancer), geom = \"polygon\", alpha = .2) + \n",
    "  guides(fill = \"none\") + \n",
    "  labs(title = \"\", x = \"Coordinate 2\", y = \"Coordinate 3\") +\n",
    "  theme_minimal() # theme_minimal(), theme_tufte() +\n",
    "  #xlim(-40,40) + ylim(-35,35) # will need to match based on whatever the best results look like\n",
    "\n",
    "ggsave(file.path(dat_dir, \"Peptide/MDS/peptide_MDS_ggplot_Coord2-3.pdf\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ce397e90-8d58-4822-b339-e913bbecfc5a",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Primary column keys in `Model/Protein[_impNA_pVals].txt` for `filter_` varargs.\n",
      "\n",
      "Primary file loaded: /Users/jasonheld/Manuscripts/2022_Sarcoma-Exosomes/vanTine_001_FinalDataWithControls/MQ/vanTine_001_90_samples/Protein/Model/Protein_pVals.txt\n",
      "\n",
      "\n",
      "scale_log2r = TRUE\n",
      "\n",
      "impute_na = FALSE\n",
      "\n",
      "complete_cases = FALSE\n",
      "\n",
      "No column-ordering of samples at `cluster_cols = TRUE`.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Make heatmap\n",
    "\n",
    "prnHM(\n",
    "  xmin = -2.5, \n",
    "  xmax = 2.5, \n",
    "  xmargin = 0.5,\n",
    "  na_col = \"white\",\n",
    "  border_color = \"black\",\n",
    "  annot_cols = c(\"Sensitivity\", \"Timing\",\"Cancer\",\"Age\",\"Gender\"),\n",
    "  annot_colnames = c(\"Drug Response\", \"Pre/Post Treatment\",\"Cancer\",\"Age\",\"Gender\"),\n",
    "  cluster_rows = TRUE, \n",
    "  cutree_rows = 1,\n",
    "  cutree_cols = 5,\n",
    "  show_rownames = FALSE, \n",
    "  show_colnames = TRUE, \n",
    "  fontsize_row = 3, \n",
    "  cellwidth = 14, \n",
    "  filter_ = exprs(count_nna >= 60),\n",
    "  filter_ = exprs(prot_n_pep >= 2),\n",
    "  hc_method_rows = \"ward.D2\", \n",
    "  hc_method_cols = \"ward.D2\", \n",
    "  p_dist_rows = 2,\n",
    "  p_dist_cols = 2,\n",
    "  clustering_distance_rows = \"manhattan\",\n",
    "  clustering_distance_cols = \"manhattan\",\n",
    "  filename = heatmap.pdf,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b4c3f392-3c34-47ea-aa22-1a0462fdcc18",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c3c81aa8-dd40-4fe2-8ffb-387832d2f879",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.2.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
